dataclients/kubernetes/ingressdefinitions.go

```diff
@@ -65,6 +65,145 @@ func (s service) getTargetPortByValue(p int) (*definitions.BackendPort, bool) {
 	return nil, false
 }
 
+// There are [1..N] Kubernetes endpointslices created for a single Kubernetes service.
+// Kubernetes endpointslices of a given service can have duplicates with different states.
+// Therefore Kubernetes endpointslices need to be de-duplicated before usage.
+// The business object skipperEndpointSlice is a de-duplicated endpoint list that concats all endpointslices of a given service into one slice of skipperEndpointSlice.
+type skipperEndpointSlice struct {
+	Meta      *definitions.Metadata
+	Endpoints []*skipperEndpoint
+	Ports     []*endpointSlicePort
+}
+
+// Conditions have to be evaluated before creation
+type skipperEndpoint struct {
+	Address string
+	Zone    string
+}
+
+func (eps skipperEndpointSlice) getPort(protocol, pName string, pValue int) int {
```

Comment:
Lets use pointer receiver following https://dave.cheney.net/2016/03/19/should-methods-be-declared-on-t-or-t

Reply:
But we use value receiver in all the other functions, too (service, endpoints). Do you mean I should change all?
To have only one different from the other makes no sense IMO.

Reply:
Up to you

Reply:
done